src/main/scala/io/iohk/ethereum/consensus/ethash/validators/EthashBlockHeaderValidator.scala

```diff
@@ -22,15 +30,14 @@ class EthashBlockHeaderValidator(blockchainConfig: BlockchainConfig) extends Blo
   protected def difficulty: DifficultyCalculator = new EthashDifficultyCalculator(blockchainConfig)
 
   def validateEvenMore(blockHeader: BlockHeader, parentHeader: BlockHeader): Either[BlockHeaderError, BlockHeaderValid] =
-    validatePoW(blockHeader)
-
-  /**
-   * Validates [[io.iohk.ethereum.domain.BlockHeader.nonce]] and [[io.iohk.ethereum.domain.BlockHeader.mixHash]] are correct
-   * based on validations stated in section 4.4.4 of http://paper.gavwood.com/
-   *
-   * @param blockHeader BlockHeader to validate.
-   * @return BlockHeader if valid, an [[io.iohk.ethereum.consensus.validators.BlockHeaderError.HeaderPoWError]] otherwise
-   */
+    if (shouldSkipPoW) Right(BlockHeaderValid) else validatePoW(blockHeader)
```

Comment:
I am not sure that ets test-s should have such impact on our production code. Have you tried maybe creating sub-class of `EthashBlockHeaderValidator` which always validate  to `Right(BlockHeaderValid)`,  and then passing it to `EthashValidators `  as parametr in `apply` ? 
How much more work would it be  ? ( just asking as maybe it to much hassle to do that, becouse little mess in class dependencies becouse of swapable consensus code)

Reply:
I'll check it out